### PR TITLE
Remove some obsolete BWC version checks

### PR DESCRIPTION
**Pull request by danhermann**

### CODE REVIEW

**Comment by danhermann** [comment]
Manually pinging @elastic/es-data-management since this spans a bunch of features 🙂

File: server/src/main/java/org/elasticsearch/cluster/metadata/DataStreamMetadata.java [hunk 1]
```diff
@@ -64,8 +64,6 @@
         }, DATA_STREAM_ALIASES);
     }
 
-    public static final Version DATA_STREAM_ALIAS_VERSION = Version.V_7_14_0;
```
Comment (joegallo) [inline server/src/main/java/org/elasticsearch/cluster/metadata/DataStreamMetadata.java:67]: Ah, I see you have an eye for detail. 👍

**Comment by danhermann** [comment]
@elasticmachine update branch

**Comment by danhermann** [comment]
@elasticmachine update branch

**Comment by joegallo** [comment]
@elasticmachine update branch

**Review by joegallo — APPROVED** [review]
+1, merge away if Jenkins approves

**Comment by danhermann** [comment]
Thank you, @joegallo!

**Pull request merged**